R/blueprint.R

```diff
@@ -26,6 +26,9 @@
 #' prediction time? This information is used by the `clean` function in the
 #' `forge` function list, and is passed on to [scream()].
 #'
+#' @param composition Either "tibble", "matrix", or "dgCMatrix" for the format
```

Comment:
I haven't used Matrix very much, but I know they have a ton of class types. Is it always a `dgCMatrix`? The `Matrix()` function can return a number of different things, and I'm not sure if setting `sparse = TRUE` always guarantees a dgCMatrix class. If not, should the option just be `"sparse_matrix"`?

Reply:
I am pretty sure we should stick with the dgCMatrix class only. I just spent some time looking through the Matrix docs, and the dgCMatrix class is a compressed, sparse, column-oriented format which is the right kind for modeling. From the Matrix docs:

> `dgCMatrix` is the “standard” class for sparse numeric matrices in the **Matrix** package.

I haven't ever seen any of the other classes used in the wild in modeling, and things like the [dfm](https://github.com/quanteda/quanteda/blob/master/R/dfm-classes.R) type in quanteda inherit from dgCMatrix.

(The DocumentTermMatrix in tm inherits from the simple triplet matrix in slam, but I don't think we need to worry about that here.)